model/map-jpa/src/main/java/org/keycloak/models/map/storage/jpa/user/entity/JpaUserEntity.java

```diff
@@ -232,16 +230,19 @@ public void setRealmId(String realmId) {
         this.metadata.setRealmId(realmId);
     }
 
+    /**
+     * @return User's username with respecting letter case.
+     */
     @Override
     public String getUsername() {
-        if (this.isMetadataInitialized()) return this.metadata.getUsername();
-        return this.username;
+        if (this.isMetadataInitialized()) return this.metadata.getUsernameWithCase();
```

Comment:
Will this work with entityversion 1?

Reply:
It should work now when the migration of the entity was fixed in the last force-push. Each entity loaded from the database should have the usernameWithCase field filled. Or am I missing something?

Reply:
The migration fixed the functional part when the username is loaded from JSON, thank you.
Will the V1 have the `this.usenameWithCase` field filled when loading the value from column directly in the next line?

Reply:
I see, it won't. I'll add fallback to `username` filed in case of `usernameWithCase` is null.